File: 구름 알고리즘 미션/그래프 탐색(1)/영상처리/영상처리.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, M] = input[0].split(" ").map(Number);
const picture = Array.from({length: M}, () => []);
const visited = Array.from({length: M}, () => Array(N).fill(false));
const directions = [[-1, 0], [0, -1], [1, 0], [0, 1]];
for(let i = 1 ; i <= M ; i++){
	picture[i-1] = input[i].split("");
}

const bfs = (x, y) => {
	let extent = 1;
	visited[x][y] = true;
	let q = [[x, y]];
	
	while(q.length > 0){
		const [cx, cy] = q.shift();
		for(const [dx, dy] of directions){
			const [mx, my] = [cx+dx, cy+dy];
			if(0 <= mx && mx < M && 0 <= my && my < N && picture[mx][my] === '#' && visited[mx][my] === false){
				q.push([mx, my]);
				visited[mx][my] = true;
				extent++;
			}
		}
	}
	
	return extent
}

let cnt = 0;
let maxBlockSize = 0;
for(let i = 0 ; i < M ; i++){
	for(let j = 0 ; j < N; j++){
		if(picture[i][j] === '#' && visited[i][j] === false){
			cnt++;
			maxBlockSize = Math.max(maxBlockSize, bfs(i, j));
		}
	}
}

console.log(cnt);
console.log(maxBlockSize);

File: 구름 알고리즘 미션/그래프 탐색(1)/영상처리/영상처리.py
from collections import deque
N, M = map(int, input().split())
picture = [list(map(str, input().rstrip())) for _ in range(M)]
directions = [[-1, 0], [0, -1], [1, 0], [0, 1]]
visited = [[False] * N for _ in range(M)]

def BFS(x, y):
	extent = 1
	q = deque([[x, y]])
	visited[x][y] = True
	
	while q:
		cx, cy = q.popleft()
		for dx, dy in directions:
			mx, my = cx + dx, cy + dy
			if 0 <= mx < M and 0 <= my < N and picture[mx][my] == '#' and not visited[mx][my]:
					q.append([mx, my])
					visited[mx][my] = True
					extent += 1
		
	return extent;
	
	
cnt = 0
max_block_size = 0
for i in range(M):
	for j in range(N):
		if picture[i][j] == '#' and not visited[i][j]:
			cnt += 1
			max_block_size = max(max_block_size, BFS(i, j))

print(cnt)
print(max_block_size)